### PR TITLE
Asynchronous FIFO updates

### PR DESCRIPTION
**Pull request by tomroberts-lowrisc**
Various lint and style updates plus some extra parameterization to allow the module to be used with Depth < 3. Also added some assertions to confirm gray coding works correctly.

### CODE REVIEW

File: hw/ip/prim/rtl/prim_fifo_async.sv
Comment (rswarbrick) [inline hw/ip/prim/rtl/prim_fifo_async.sv]: Super nit, but if you're touching this anyway maybe the terms in the concatenation need spaces between them?

File: hw/ip/prim/rtl/prim_fifo_async.sv
Comment (rswarbrick) [inline hw/ip/prim/rtl/prim_fifo_async.sv:32]: Maybe the commit message should be "Depth <= 2"? Since we require a power of 2, maybe it's better to pretend numbers like 3 don't exist :-p

Also, do we still need to check that Depth > 0?

Comment (tomroberts-lowrisc) [reply hw/ip/prim/rtl/prim_fifo_async.sv:32]: Will amend - not sure about checking for Depth > 0 - not something a user would do accidentally

**Comment by tjaychen** [comment]
@tomroberts-lowrisc just wanted to check-in and see if you're okay to merge this one?

**Comment by tjaychen** [comment]
o wait. maybe you were waiting for an approval from me. sorry.

File: hw/ip/prim/rtl/prim_fifo_async.sv
Comment (tjaychen) [inline hw/ip/prim/rtl/prim_fifo_async.sv:126]: will the depth = 1 case complain here? seems like we'll be trying to concatenate 0 of 1'b0

Comment (rswarbrick) [reply hw/ip/prim/rtl/prim_fifo_async.sv:126]: I think a zero-count replication is explicitly allowed in the spec (paragraph 5 of section 11.4.12.1), provided it's inside a concatenation (which this is).

Comment (tomroberts-lowrisc) [reply hw/ip/prim/rtl/prim_fifo_async.sv:126]: Agreed, we do this in various other places

Comment (tjaychen) [reply hw/ip/prim/rtl/prim_fifo_async.sv:126]: wow, honestly i'm surprised some tool somewhere didn't barf on this (but i just noticed in ibex we have something similar).
This looks good to me. Should we merge?

**Review by tjaychen — APPROVED** [review]

**Pull request merged**
